presto-orc/src/main/java/com/facebook/presto/orc/TupleDomainFilter.java

```diff
@@ -0,0 +1,966 @@
+        public boolean testBoolean(boolean value)
+        {
+            return this.value == value;
+        }
+
+        @Override
+        public int hashCode()
+        {
+            return Objects.hash(value, nullAllowed);
+        }
+
+        @Override
+        public boolean equals(Object obj)
+        {
+            if (this == obj) {
+                return true;
+            }
+
+            if (obj == null || getClass() != obj.getClass()) {
+                return false;
+            }
+
+            BooleanValue other = (BooleanValue) obj;
+            return this.value == other.value &&
+                    this.nullAllowed == other.nullAllowed;
+        }
+
+        @Override
+        public String toString()
+        {
+            return toStringHelper(this)
+                    .add("value", value)
+                    .add("nullAllowed", nullAllowed)
+                    .toString();
+        }
+    }
+
+    class BigintRange
+            extends AbstractTupleDomainFilter
+    {
+        private final long lower;
+        private final long upper;
+
+        public BigintRange(long lower, long upper, boolean nullAllowed)
+        {
+            super(nullAllowed);
+            checkArgument(lower <= upper, "lower must be less than or equal to upper");
+            this.lower = lower;
+            this.upper = upper;
+        }
+
+        @Override
+        public boolean testLong(long value)
+        {
+            return value >= lower && value <= upper;
+        }
+
+        public long getLower()
+        {
+            return lower;
+        }
+
+        public long getUpper()
+        {
+            return upper;
+        }
+
+        public boolean isSingleValue()
+        {
+            return upper == lower;
+        }
+
+        @Override
+        public boolean equals(Object o)
+        {
+            if (this == o) {
+                return true;
+            }
+
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+
+            BigintRange that = (BigintRange) o;
+            return lower == that.lower &&
+                    upper == that.upper &&
+                    nullAllowed == that.nullAllowed;
+        }
+
+        @Override
+        public int hashCode()
+        {
+            return Objects.hash(lower, upper, nullAllowed);
+        }
+
+        @Override
+        public String toString()
+        {
+            return toStringHelper(this)
+                    .add("lower", lower)
+                    .add("upper", upper)
+                    .add("nullAllowed", nullAllowed)
+                    .toString();
+        }
+    }
+
+    class BigintValues
+            extends AbstractTupleDomainFilter
+    {
+        private static final long EMPTY_MARKER = 0xdeadbeefbadefeedL;
+        // from Murmur hash
+        private static final long M = 0xc6a4a7935bd1e995L;
+
+        private final long[] originalValues;
+        private final long[] longs;
```

Comment:
nit: Is `originalValues` values in an array, and `longs` are these values but present in hash table? -- In that case maybe call `longs` as `hashTable`? :)